src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -40,6 +37,9 @@ public sealed class GrpcChannel : ChannelBase, IDisposable
     {
         internal const int DefaultMaxReceiveMessageSize = 1024 * 1024 * 4; // 4 MB
 
+        private readonly ConcurrentDictionary<IMethod, GrpcMethodInfo> _methodInfoCache;
```

Comment:
Could you be more specific?

I cached the delegate used with GetOrAdd in a field `_createMethodInfoFunc` so it is allocated once per channel, rather than once per call. Other than that I didn't notice additional allocations.